src/MCPClient/lib/clientScripts/move_or_merge.py

```diff
@@ -1,8 +1,72 @@
 #!/usr/bin/env python2
 
+import errno
+import filecmp
 import os
 import shutil
 
+import scandir
+
+
+def mkdir_p(path):
+    """Create a directory if it doesn't already exist."""
+    try:
+        os.makedirs(path)
+    except OSError as err:
+        if err.errno == errno.EEXIST:
+            pass
+        else:
+            raise
+
+
+def _move_file(src, dst):
+    """
+    Move an individual file from ``src`` to ``dst``.
+    """
+    # Ensure the destination directory exists
+    mkdir_p(os.path.dirname(dst))
+
+    # If the file already exists at the destination, check if it's the same.
+    # If so, we can clean up the original file and we're done.  If not, we're
+    # at risk of losing data, so error out.
+    if os.path.isfile(dst):
+        if filecmp.cmp(src, dst):
+            os.unlink(src)
+        else:
+            raise RuntimeError(
+                "Tried to move src=%s to dst=%s, but dst exists and is different"
+                % (src, dst)
+            )
+    else:
+        shutil.move(src, dst)
+
+
+def move_or_merge(src, dst):
+    """
+    Move a file/directory to a new location, or merge two directories.
+
+    If ``dst`` doesn't exist, it's a simple move: ``src`` is moved to the same
+    path as ``dst``.
```

Comment:
The documentation is really nice around this PR. Do you have internal guidelines on this? I feel like it's something we haven't yet found consistency on, so any tips on bringing that consistency into the work would be great!

Reply:
@ross-spencer Agreed - all credit to @alexwlchan for that :-)